include/fastdds/rtps/history/IPayloadPool.h

```diff
@@ -22,27 +22,76 @@
 #include <fastdds/rtps/common/CacheChange.h>
 
 #include <cstdint>
-#include <functional>
+#include <memory>
 
 namespace eprosima {
 namespace fastrtps {
 namespace rtps {
 
+/**
+ * An interface for classes responsible of serialized payload management.
+ */
 class IPayloadPool
 {
 public:
 
     virtual ~IPayloadPool() = default;
 
+    /**
+     * @brief Get a serialized payload for a new sample.
+     * 
+     * This method will usually be called in one of the following situations:
+     * @li When a writer creates a new cache change
+     * @li When a reader receives the first fragment of a cache change
+     * 
+     * In both cases, the received @c size will be for the whole serialized payload.
+     * 
+     * @param [in]     size          Number of bytes required for the serialized payload
+     * @param [in,out] cache_change  Cache change to assign the payload to
+     * 
+     * @returns whether the operation succeeded or not
+     * 
+     * @post @li Field @c payload_owner of @c cache_change equals this
+     * @post @li Field @c serializedPayload.data points to a buffer of at least @c size bytes
+     * @post @li Field @c serializedPayload.max_size is greater than or equal to @c size
```

Comment:
```suggestion
     * @post 
     *     @li Field @c payload_owner of @c cache_change equals this
     *     @li Field @c serializedPayload.data points to a buffer of at least @c size bytes
     *     @li Field @c serializedPayload.max_size is greater than or equal to @c size
```
It gets rendered the same, but is more readable in the code itself